### PR TITLE
Adding new OpenSearch versions and updating compatibility matrix

### PR DESCRIPTION
**Pull request by VachaShah**
Signed-off-by: Vacha Shah <vachshah@amazon.com>

### Description
Adding new OpenSearch versions and updating the compatibility matrix.

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
Would merge on 🟢

**Comment by VachaShah** [comment]
Tests were failing due to a recent change in escape_utils gem. The `escape_url` method was changed to `escape_uri` according to documentation https://github.com/brianmario/escape_utils/

**Comment by dblock** [comment]
> Tests were failing due to a recent change in escape_utils gem. The `escape_url` method was changed to `escape_uri` according to documentation https://github.com/brianmario/escape_utils/

This is telling me that there's no restriction on the version of the gem that brings in `escape_uri` in Gemfile and there should be. Can fix this separately - which version introduced the change?

**Review by dblock — APPROVED** [review]

File: opensearch-api/lib/opensearch/api/utils.rb [hunk 1]
```diff
@@ -40,7 +40,7 @@ module Utils
       # @api private
       def __escape(string)
         return string if string == '*'
-        defined?(EscapeUtils) ? EscapeUtils.escape_url(string.to_s) : CGI.escape(string.to_s)
+        defined?(EscapeUtils) ? EscapeUtils.escape_uri(string.to_s) : CGI.escape(string.to_s)
```
Comment (saratvemulapalli) [inline opensearch-api/lib/opensearch/api/utils.rb:43]: Curious why we had to change to use URI ?

**Review by saratvemulapalli — APPROVED** [review]

File: opensearch-api/lib/opensearch/api/utils.rb (hunk 1 above)
Comment (saratvemulapalli) [reply opensearch-api/lib/opensearch/api/utils.rb:43]: Nvm I just saw the comment above.

**Comment by VachaShah** [comment]
> > Tests were failing due to a recent change in escape_utils gem. The `escape_url` method was changed to `escape_uri` according to documentation https://github.com/brianmario/escape_utils/
> 
> This is telling me that there's no restriction on the version of the gem that brings in `escape_uri` in Gemfile and there should be. Can fix this separately - which version introduced the change?

The change was introduced in version 1.3.0.

File: opensearch-api/lib/opensearch/api/utils.rb (hunk 1 above)
Comment (dblock) [reply opensearch-api/lib/opensearch/api/utils.rb:43]: I removed it in https://github.com/opensearch-project/opensearch-ruby/pull/74

**Comment by dblock** [comment]
> 

Removed it in https://github.com/opensearch-project/opensearch-ruby/pull/74 since it's deprecated altogether. No need for a library that monkey patches anything :)

**Pull request merged**